xml/smb_native_os.xml

```diff
@@ -384,6 +384,33 @@
       <param pos="2" name="os.build"/>
    </fingerprint>
 
+      <!-- Mac OS X -->
+   <fingerprint pattern="^Samba (3\.0\.28a-apple)$">
```

Comment:
Better, but I'm not 100% sure we want to be including the `-apple` part in the version.  There are pros and cons to doing it and not doing it.  I don't have a strong opinion either way but it is worth thinking about for the future.